### ISSUE TITLE
Main process hangs: Closing all ReceivePorts necessary to exit main process

### ISSUE THREAD

**Issue by xErik** [maxim-saplin/isolate_pool_2#3]
Hello!

I did some additional testing. 

In certain cases, the invoking main process hangs after `pool.stop()` has been called. This hanging process must be killed via CTRL+C.

The complete test case is given below: Running `dart run test.dart` does not exit the main process.

**Problem**

The issue is discussed here: https://github.com/dart-lang/sdk/issues/49083

**Solution**

I tinkered with the package's code and closed all `ReceivePort`s. The main process does not hang anymore and the program exits correctly. There is probably a more elegant way to achieve this within the package's code, my approachis simply a proof.

```dart 

final List<ReceivePort> rPorts = [];

Future start() async {
  for (var i = 0; i < numberOfIsolates; i++) {
    var receivePort = ReceivePort();
    rPorts.add(receivePort); // COLLECTING ReceivePorts
  }
}


void stop() {
  for (var i in _isolates) {
    i.kill();

    for (var rPort in rPorts) {
      rPort.close(); // CLOSING ReceivePorts
    }
}
```

**Test Code**

```dart
import 'package:isolate_pool_2/isolate_pool_2.dart';

// ---------------------------------------------------------
// MAIN
// ---------------------------------------------------------

void main(List<String> args) async {
  final List<Future> futures = [];
  final pool = IsolatePool(2);
  await pool.start();

  try {
    for (var i = 0; i < 10; i++) {
      final job = MyJob(i);
      futures.add(pool.scheduleJob(job));
    }

    await Future.wait(futures, eagerError: true);
  } on MyException catch (e) {
    print('CAUGHT: ${e.message}');
  } finally {
    try {
      print('STOPPING ...');
      pool.stop();
    } catch (_) {
      // ignore safety measure
    }
  }

  print('Hanging ...');
}

// ---------------------------------------------------------
// JOB
// ---------------------------------------------------------

class MyJob extends PooledJob {
  final int i;
  MyJob(this.i);

  @override
  job() async {
    print('JOB: $i');
    await Future.delayed(Duration(seconds: 1));

    if (i == 2) {
      throw MyException('Thrown at: $i');
    }
  }
}

// ---------------------------------------------------------
// EXCEPTION
// ---------------------------------------------------------

class MyException implements Exception {
  final String message;
  MyException(this.message);
}
```

**Comment by maxim-saplin** [comment]
Thanks a lot for the investigation! Could you please submit a PR so I could check it (a unit test + fix)?

**Comment by maxim-saplin** [comment]
I believe replicating the issue with unit test is not possible. Added the fix you suggested and published 1.0.4 to pub.dev. Thanks for the help!
